tuned/utils/config_parser.py

```diff
@@ -0,0 +1,43 @@
+# ConfigParser wrapper providing compatibility layer for python 2.7/3
+
+try:
+	python3 = True
+	import configparser as cp
+except ImportError:
+	python3 = False
+	import ConfigParser as cp
+	from StringIO import StringIO
+	import re
+
+class Error(cp.Error):
+	pass
+
+if python3:
+
+	class ConfigParser(cp.ConfigParser):
+		pass
+
+else:
+
+	class ConfigParser(cp.ConfigParser):
+
+		def __init__(self, delimiters=None, inline_comment_prefixes=None, strict=True, *args, **kwargs):
+			cp.ConfigParser.__init__(self, *args, **kwargs)
+			self._inline_comment_prefixes = inline_comment_prefixes or []
+			self._re = re.compile("\s+(%s).*" % ")|(".join(list(self._inline_comment_prefixes)))
```

Comment:
This means inline comments have to start with whitespace character.  To my mind this is not robust and this shouldn't be needed, i.e. they should start with the inline comment character itself.  Is this intentional?

Reply:
Yes, it is intentional. From the python 2.7 ConfigParser source code:
>  ';' is a comment delimiter only if it follows a spacing character

I.e. in the python 2.7 the ConfigParsr takes the ';' as the comment (it is usually used this way in the INI file). Maybe we should also use the `inline_comment_prefixes = ['#', ';']` for better compatibility?

Also with the python 3 and `inline_comment_prefixes = ['#']` it seems there has to be a white space before the comment, so for me the behavior seems consistent.

Reply:
> Yes, it is intentional. From the python 2.7 ConfigParser source code:
> 
> > ';' is a comment delimiter only if it follows a spacing character
> 
> I.e. in the python 2.7 the ConfigParser takes the ';' as the comment (it is usually used this way in the INI file). Maybe we should also use the `inline_comment_prefixes = ['#', ';']` for better compatibility?
> 
> Also with the python 3 and `inline_comment_prefixes = ['#']` it seems there has to be a white space before the comment, so for me the behavior seems consistent.

This holds only for the so called 'inline comments', the normal comments starts the line and do not need to be prefixed by the white space.